src/unix/core.c

```diff
@@ -1240,3 +1240,40 @@ int uv_translate_sys_error(int sys_errno) {
   /* If < 0 then it's already a libuv error. */
   return sys_errno <= 0 ? sys_errno : -sys_errno;
 }
+
+
+int uv_getenv(const char* name, char* buffer, size_t* size) {
+  char* var;
+  size_t len;
+
+  if (name == NULL || buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  var = getenv(name);
+
+  if (var == NULL)
+    return -ENOENT;
+
+  len = strlen(var);
+
+  if (len >= *size) {
+    *size = len + 1;
+    return -ENOBUFS;
+  }
+
+  memcpy(buffer, var, len + 1);
+  *size = len;
```

Comment:
Yea, last year we decided on:

- On `UV_ENOBUFS` set `size` to the required space, including the null.
- On success, set `size` to the string length and null terminate the string.

I don't know if there are functions that still don't follow this, but there were at least a few PRs to migrate the codebase toward this.